portfolio/src/main/java/com/google/sps/servlets/AddCommentServlet.java

```diff
@@ -34,15 +45,52 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
     Date postDate = new Date();
     String text = request.getParameter("comment-text");
 
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "image");
+
     // Create comment entity 
     Entity commentEntity = new Entity("BlogComment");
     commentEntity.setProperty("text", text);
     commentEntity.setProperty("postDate", postDate);
+    commentEntity.setProperty("imageURL", imageUrl);
 
     // Upload comment to Datastore 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/blog.html");
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```

Comment:
We can also mark this method as nullable

Reply:
Sure thing